### ISSUE TITLE
taylor returns one order less than given

### ISSUE THREAD

**Issue by vitaminace33** [gnu-octave/symbolic#1303]
The order of a [Taylor polynomial](https://en.wikipedia.org/wiki/Taylor%27s_theorem) is (generically) the degree of the polynomial. OctSymPy's `taylor` function returns a polynomial of one degree less than the given order, as clearly shown in [the reference's examples](https://octave.sourceforge.io/symbolic/function/@sym/taylor.html)! The authors of the function may have confused the order with the number of terms (which is one more than the order). Please, fix the disorder, may Brook Taylor rest in peace.

**Comment by cbm755** [comment]
### TL-DR;

I think the code is doing what we want, but we definitely want to fix up the docs.

### Details:

I agree there is a problem here.

The underlying library SymPy does this:
```
var('x')
f = exp(x)
f.series(x, 0, 6)

         2    3    4     5        
        x    x    x     x     ⎛ 6⎞
1 + x + ── + ── + ── + ─── + O⎝x ⎠
        2    6    24   120        
```
Note:
   * the SymPy command is **not** called "Taylor series"
   * the docs are not very precise about this "The number of terms upto which the series is to be expanded." [https://docs.sympy.org/latest/modules/series/series.html], which is vague.
      - elsewhere it says "The value used to represent the order in terms of x**n, up to which the series is to be expanded."   [https://docs.sympy.org/latest/modules/core.html#sympy.core.expr.Expr.series]

So I think its clear enough that SymPy is using something like "the order of the truncation error".

Next, I checked Matlab's Symbolic Toolbox (we generally want to be compatible with that).  They refer to "truncation order", a term I'm not familiar but seems to agree with what we're doing.
Maple seems to call it the "expansion order" [https://www.maplesoft.com/support/help/Maple/view.aspx?path=taylor].  [Elsewhere in their docs](https://www.maplesoft.com/support/help/maple/view.aspx?path=mtaylor) Maple uses the term "truncation order".

**Comment by vitaminace33** [comment]
Indeed, Matlab's documentation states:

> Use Order to control the truncation order. For example, approximate the same expression up to the orders 7 and 9.
```
T8 = taylor(f,x,'Order',8);
T10 = taylor(f,x,'Order',10);
```

Maple's documentation states:

> The taylor command computes the order n Taylor series expansion

which makes no sense, a series is an infinite sum, a Taylor series is a power series, a sum of infinite powers and, hence, has no defined "order". The should say `the order n-1 Taylor polynomial`.

Perhaps they refer to the order of the approximation term to relate to numerical methods which are usually referred by their order of approximation, e.g. Runge-Kutta Iv.

**Comment by vitaminace33** [comment]
The docs should definitively be clear on this.

**Comment by cbm755** [comment]
Would you like to have a go at editing our docs?  The file is `inst/@sym/taylor.m`.

**Comment by vitaminace33** [comment]
I have two options:

```
     The order of the remainder term is set with a key/value pair which
     defaults to 6:

     The degree of the polynomial is controlled via the order of the
     remainder with a key/value pair which defaults to 6:
          syms x
          f = exp(x);
          p = taylor(f, x, 0, 'order', 6)
            ⇒ p = (sym)
                  5    4    3    2
                 x    x    x    x
                ─── + ── + ── + ── + x + 1
                120   24   6    2
     In the above example, using big O notation, ‘f - p = O(x^6)’.
```

What do you think?

**Comment by cbm755** [comment]
Second one?  

Maybe we can say it explicitly as well, if not precisely: "As we see in the previously example, the degree of the returned Taylor polynomial will be (at most) one less than `order`"

**Comment by cbm755** [comment]
Anyway, send an edit and we can bikeshed it together ;-)

**Comment by vitaminace33** [comment]
On it... although I don't know what "to bikeshed it together" means.

**Comment by cbm755** [comment]
Fixed by #1305.
